integration/kubetoken/kubetoken_test.go

```diff
@@ -0,0 +1,54 @@
+//go:build integration
+// +build integration
+
+// Copyright 2023 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kubetoken
+
+import (
+	"encoding/json"
+	"github.com/golang-jwt/jwt/v4"
+	"github.com/okteto/okteto/integration"
+	"github.com/okteto/okteto/integration/commands"
+	"github.com/okteto/okteto/pkg/okteto"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+// TestKubetokenHasExpiration test that kubernetes returns a dynamic token with a set expiration
+func TestKubetokenHasExpiration(t *testing.T) {
+	t.Parallel()
+
+	oktetoPath, err := integration.GetOktetoPath()
```

Comment:
We'll need to ensure that the cluster is on 1.27 or later and that the okteto cluster has the endpoint. We should skip this tests if those criteria are not meet

Reply:
@jLopezbarb don't we run our tests against staging? I assume that if we have control over the target cluster, we can safely assume this feature will be always available. Please correct me if I am wrong

Reply:
@andreafalzetti  Technically, they can run against any cluster changing properly some environment variable. But check my previous comment. Could you check since which Kubernetes version is TokenRequest API available?

Reply:
@ifbyol the TokenRequest API seems to be available in Kubernetes since 1.14 according to [this](https://jpweber.io/blog/a-look-at-tokenrequest-api/)